### PR TITLE
Fix the mocked http server port in use issue

### PR DESCRIPTION
**Pull request by wezhang**
Instead of fixed http port confliced in concurrent auto testing,
used dynamic port for mocked http server and Groovy templete
to evaluate the port value into the faked http response body.

Also introduced Kotlin classes for testing

Signed-off-by: Zhang Wei <zhwe@microsoft.com>

### CODE REVIEW

**Comment by wezhang** [comment]
to fix #819

**Comment by wezhang** [comment]
```
[ERROR] COMPILATION ERROR : 
[ERROR] /home/travis/build/Microsoft/azure-tools-for-java/Utils/hdinsight-node-common/Test/java/com/microsoft/azure/hdinsight/spark/jobs/JobUtilsScenario.java:[24,50] cannot find symbol
  symbol:   class MockHttpService
  location: package com.microsoft.azure.hdinsight.spark.common
```

CI server issue that not found in local

**Comment by adashen** [comment]
It seems new dependency is introduced. We cannot merge this PR. New dependency should be checked and update thirdparty notice

**Comment by Eskibear** [comment]
retest this please

File: Utils/hdinsight-node-common/pom.xml [hunk 1]
```diff
@@ -187,5 +239,14 @@
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains.kotlin</groupId>
+            <artifactId>kotlin-stdlib-jre8</artifactId>
```
Comment (Eskibear) [inline Utils/hdinsight-node-common/pom.xml]: test scope

Comment (wezhang) [inline Utils/hdinsight-node-common/pom.xml]: I suppose the Kotlin lang support is not only for test code, but also for the whole project

Comment (wezhang) [inline Utils/hdinsight-node-common/pom.xml]: test first

**Comment by wezhang** [comment]
Already checked the output in local:

```bash
 ./gradlew dependencies --project-dir ./PluginsAndFeatures/azure-toolkit-for-intellij --configuration compile -q | grep kotlin
```

No kotlin in dependency, but still got: 

```diff
+ ./gradlew dependencies --project-dir ./PluginsAndFeatures/azure-toolkit-for-intellij --configuration compile -q
+ diff /home/jenkins/dependencyLog//develop/intellij/309-intellij.log /home/jenkins/dependencyLog//PR/intellij/937-intellij.log
+ cat
+ cat /home/jenkins/dependencyLog//patch/PR-937.patch
+ wc -l
+ [ 15 -eq 0 ]
+ cat /home/jenkins/dependencyLog//patch/PR-937.patch
48a49
>    com.microsoft.azuretools.wap4ej.build:com.microsoft.azuretools.sdk:jar:dependencies/annotations-13.0.jar:3.0.7.qualifier:system
111a113,115
>    com.microsoft.azuretools.wap4ej.build:com.microsoft.azuretools.sdk:jar:dependencies/kotlin-stdlib-1.1.3-2.jar:3.0.7.qualifier:system
>    com.microsoft.azuretools.wap4ej.build:com.microsoft.azuretools.sdk:jar:dependencies/kotlin-stdlib-jre7-1.1.3-2.jar:3.0.7.qualifier:system
>    com.microsoft.azuretools.wap4ej.build:com.microsoft.azuretools.sdk:jar:dependencies/kotlin-stdlib-jre8-1.1.3-2.jar:3.0.7.qualifier:system
208c208,213
< |    \--- org.apache.commons:commons-lang3:3.5
---
> |    +--- org.apache.commons:commons-lang3:3.5
> |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.1.3-2
> |         +--- org.jetbrains.kotlin:kotlin-stdlib:1.1.3-2
> |         |    \--- org.jetbrains:annotations:13.0
> |         \--- org.jetbrains.kotlin:kotlin-stdlib-jre7:1.1.3-2
> |              \--- org.jetbrains.kotlin:kotlin-stdlib:1.1.3-2 (*)
+ echo ALERT: New dependencies introduced!
ALERT: New dependencies introduced!
```

@Eskibear could you hlep to check it? Thanks!

**Comment by voidfunction** [comment]
@wezhang the Kotlin dependencies may be referenced in Eclipse project. Please confirm it.

**Comment by wezhang** [comment]
Sending a new PR #939 fixed the build issue, looks like there might be some pollution in build server global repository.

@Eskibear , does the local maven repository ` -Dmaven.repo.local=$SCRIPTPATH/.repository` need to be added into the dependency build?

```
 mvn dependency:list -f ./PluginsAndFeatures/azure-toolkit-for-eclipse/pom.xml -DoutputFile=target/eclipse.log -Dsort=true
```

**Comment by Eskibear** [comment]
@wezhang It's due to the dirty `PR-###.patch` file. Already fixed in jenkins dependency-check script.

**Comment by wezhang** [comment]
Thanks @Eskibear !

File: Utils/hdinsight-node-common/pom.xml [hunk 2]
```diff
@@ -81,8 +81,45 @@
...
+                <executions>
+                    <!-- Replacing default-testCompile as it is treated specially by maven -->
+                    <execution>
+                        <id>default-testCompile</id>
+                        <phase>none</phase>
+                    </execution>
+                    <execution>
+                        <id>java-test-compile</id>
+                        <phase>test-compile</phase>
+                        <goals> <goal>testCompile</goal> </goals>
```
Comment (djlan) [inline Utils/hdinsight-node-common/pom.xml]: ><goals> <goal>testCompile</goal> </goals> [](start = 24, length = 41)

Format this?

Comment (wezhang) [inline Utils/hdinsight-node-common/pom.xml]: An XML inline simple format for the single element.  😄

**Pull request merged**